### PR TITLE
Added service routing support

### PR DESCRIPTION
**Pull request by YourTechBud**

### CODE REVIEW

**Comment by YourTechBud** [comment]
Fixes #769

**Comment by YourTechBud** [comment]
Tested everything it works. @sharadregoti please review

**Comment by sharadregoti** [comment]
tmr

File: gateway/config/routing.go [hunk 1]
```diff
@@ -16,14 +16,14 @@ type Route struct {
 }
 
 // SelectTarget returns a target based on the weights assigned
-func (r *Route) SelectTarget(weight float64) (RouteTarget, error) {
+func (r *Route) SelectTarget(weight int32) (RouteTarget, error) {
 
 	// Generate a random float in the range 0 to 1 if provided weight in lesser than zero
```
Comment (sharadregoti) [inline gateway/config/routing.go]: change comment

File: runner/server/handle.go [hunk 2]
```diff
@@ -234,6 +234,74 @@ func (s *Server) HandleApplyEventingService() http.HandlerFunc {
...
+	return func(w http.ResponseWriter, r *http.Request) {
+		defer utils.CloseTheCloser(r.Body)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+
+		// Verify token
+		_, err := s.auth.VerifyToken(utils.GetToken(r))
+		if err != nil {
+			logrus.Errorf("Failed to apply service - %s", err.Error())
```
Comment (sharadregoti) [inline runner/server/handle.go]: change the log here

File: runner/utils/driver/istio/apply.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+		service.Scale.Concurrency = 50
+	}
+
+	// Adjust the min scale in case of tcp based services. Min scale for tcp services need to be at least 1.
+	adjustMinScale(service)
+
+	// TODO: remove artifact store related code
+	token, err := i.auth.GenerateTokenForArtifactStore(service.ID, service.ProjectID, service.Version)
+	if err != nil {
+		return err
```
Comment (sharadregoti) [inline runner/utils/driver/istio/apply.go]: log the errors here using logrus

Comment (YourTechBud) [reply runner/utils/driver/istio/apply.go]: I'm logging the error here because it eventually gets logged in the handler

**Pull request merged**